OriginProtocol/origin
# Fractional usage UI - Initial Work

### Checklist:

- [x] Test your work and double check you didn't break anything

### Description:
This is finally ready for review. 

Apologies to the reviewers for the size of this PR - I kept wanting to merge but I didn't want to submit anything that was truly "broken" since this can't be easily hidden once it's released, and I'm just now getting everything pretty much working - with quite a few caveats. 

#### This PR includes:
- Seller's calendar (daily calendar for Housing and hourly calendar for Services)
- Buyer's calendar (visible on listing detail pages)
- Seller's "edit listing" workflow (currently only accessible via an "edit" button on listing detail pages)
- Updates to schemas

I'm making progress toward matching [Aure's designs](https://app.zeplin.io/project/59fa2311bac7acbc8d953da9?seid=5b1826bd16e713444156ae61) (which are still a WIP) but some styling stuff is hard and time consuming in the calendar components, so it's not pixel perfect yet.

#### Testing Hints
- Create a Housing listing (notice daily availability calendar)
- Create a Services listing (notice hourly availability calendar)
- As a seller, view the listings you created and use the new "edit" button to update a listing
- As a buyer, view the listings you created and book a time slot using the calendar on the Listing Detail page

Here's my current fractional usage to-do list, which I'll continue working through in other (smaller) PRs:

- /update/:listingId route should check to make sure user is listing’s seller before rendering for security purposes
- seller’s calendar - pressing back button breaks calendar functionality
- ~~buyer’s hourly calendar - show price of each slot in the slot~~
- ~~seller’s hourly (and maybe daily) calendar shouldn’t allow user to create overlapping events~~ 
- ~~seller’s hourly calendar - hide price and show “unavailable” if slot is unavailable~~
- ~~implement recurring events for events that occur at the same day/time every week~~
- ~~in seller’s view, using date dropdowns creates duplicate event~~
- seller’s steps are different for fractional listings
- buyer’s steps may be different for fractional listings
- localization of calendar strings
- ~~seller’s hourly calendar - add dropdown menus to change selected start/end dates~~
- seller’s calendar - implement custom radio button styles for availability toggle
- ~~make hourly calendar scroll to about 8am on page load instead of showing midnight - 9am~~
- seller’s hourly calendar - customize styling of data select carat per zeplin
- buyer’s and seller’s calendar - pressing enter key while dates are selected should “submit” form
- buyer’s calendar - style “unavailable” message to look more like error message
- ~~buyer’s calendar - clicking “cancel” should remove highlighting from selected cells~~
- add hover state to calendar time slots to emphasize interactive features
- add link to "edit listing" workflow on "my listings" list
